zerver/lib/notification_data.py

```diff
@@ -19,6 +19,31 @@ def __post_init__(self) -> None:
         if self.wildcard_mention_notify:
             assert "wildcard_mentioned" in self.flags
 
+    @classmethod
+    def from_lists(
+        cls,
+        user_id: int,
+        flags: List[str],
+        online_push_user_ids: List[int],
+        stream_push_user_ids: List[int],
+        stream_email_user_ids: List[int],
+        wildcard_mention_user_ids: List[int],
+        muted_sender_user_ids: List[int],
+    ) -> "UserMessageNotificationsData":
```

Comment:
This function has a dangerous interface, in that calling it in a loop will take quadratic time.  We really want to accept `Set` objects as what is passed into this.

Reply:
I think that once we fix this, we can just pass these parameters from the `send_request` object directly without converting Sets to Lists.

Reply:
Changed to take in `Set`s